src/main/java/uk/gov/hmcts/reform/sscs/util/PdfRequestUtil.java

```diff
@@ -128,21 +132,25 @@ private static void handlePostHearing(SscsCaseData sscsCaseData) {
     }
 
 
-    protected static void setRequestDetailsForPostHearingType(SscsCaseData sscsCaseData) {
+    protected static String getRequestDetailsForPostHearingType(SscsCaseData sscsCaseData) {
         PostHearingRequestType postHearingRequestType = sscsCaseData.getPostHearing().getRequestType();
-        switch (postHearingRequestType) {
-            case SET_ASIDE:
-                requestDetails = sscsCaseData.getDocumentGeneration().getBodyContent();
-                break;
-            case CORRECTION:
-                requestDetails = sscsCaseData.getDocumentGeneration().getCorrectionBodyContent();
-                break;
-            case STATEMENT_OF_REASONS:
-            case PERMISSION_TO_APPEAL:
-            case LIBERTY_TO_APPLY:
-            default:
-                throw new IllegalArgumentException("handlePostHearing has unexpected postHearingRequestType: " + postHearingRequestType);
+
+        if (nonNull(postHearingRequestType)) {
+            switch (postHearingRequestType) {
+                case SET_ASIDE:
+                    return sscsCaseData.getDocumentGeneration().getBodyContent();
+                case CORRECTION:
+                    return sscsCaseData.getDocumentGeneration().getCorrectionBodyContent();
+                case STATEMENT_OF_REASONS:
+                    return sscsCaseData.getDocumentGeneration().getStatementOfReasonsBodyContent();
+                case PERMISSION_TO_APPEAL:
+                case LIBERTY_TO_APPLY:
+                default:
+                    break;
+            }
         }
+
+        throw new IllegalArgumentException("handlePostHearing has unexpected postHearingRequestType: " + postHearingRequestType);
```

Comment:
I don't think this change is required. The exception should be in the default case, and null type will be covered. Remember the exception needs to be thrown if one of the remaining post hearing types is implemented, but this method is not updated appropriately.

Reply:
done